apps/wallet/src/modals/QrScan.tsx

```diff
@@ -0,0 +1,186 @@
+import type { FC } from 'react';
+import { useEffect, useMemo, useState } from 'react';
+import type { LayoutChangeEvent } from 'react-native';
+import {
+	ActivityIndicator,
+	Image,
+	StyleSheet,
+	TouchableOpacity,
+} from 'react-native';
+import {
+	Camera,
+	useCameraDevice,
+	useCameraPermission,
+	useCodeScanner,
+} from 'react-native-vision-camera';
+import type { Networks } from '@walless/core';
+import type { ModalConfigs } from '@walless/gui';
+import { modalActions, Text, View } from '@walless/gui';
+import { QrFrame, Times } from '@walless/icons';
+import assets from 'utils/assets';
+import { getNetworkInfo } from 'utils/helper';
+
+import { ModalId } from './types';
+
+interface QRScannerProps {
+	onScan: (code: string) => void;
+	network: Networks;
+}
+
+type Props = QRScannerProps & {
+	config: ModalConfigs;
+};
+
+const QRScanner: FC<Props> = ({ config, network, onScan }) => {
+	const [layoutHeight, setLayoutHeight] = useState(200);
+	const [layoutWidth, setLayoutWidth] = useState(200);
+
+	const { hasPermission, requestPermission } = useCameraPermission();
+	const [active, setActive] = useState(false);
+	const device = useCameraDevice('back');
+
+	const networkInfo = useMemo(() => getNetworkInfo(network), [network]);
+
+	if (!hasPermission) {
+		requestPermission();
+	}
```

Comment:
should we move this into `useEffect`?

Reply:
actually we can make a custom effect that wrap `useCameraPermission`, inside that custom hook, we'll cover `requestPermission`, e.g:
```typescript
const useVisionCamera = (cameraType: string) => {
  const { hasPermission, requestPermission } = useCameraPermission();
  const camera = useCameraDevice(cameraType);

  useEffect(() => {
    if (!hasPermission) {
      requestPermission();
    }
  }, [hasPermission]);

  return { camera, hasPermission };
};

// then use it inside component:
const { camera, hasPermission } = useVisionCamera('back');
```